src/state/slices/portfolioSlice/portfolioSliceCommon.ts

```diff
@@ -18,7 +18,7 @@ export type PortfolioAccounts = {
 
 // aggregated balances across all accounts in a portfolio for the same asset
 // balance in base units of asset
-export type AssetBalancesById = Record<AssetId, string>
+export type AssetBalancesById = PartialRecord<AssetId, string>
```

Comment:
guts of the PR - this was never correct. all other changes are downstream from this change.

consumers were incorrectly expecting values to be defined strings, when they could indeed be undefined.

all downstream consumption has been updated to wrap values in `bnOrZero` accordingly